File: frontend/src/features/leads/api/dtos/lead-details.dto.ts
import type {LeadStatus} from "@/entities/leads/model/types";

export type LeadDetailsDto = {
  id: number;
  propertyId: number;
  fullName: string;
  email?: string;
  phoneNumber?: string;
  message?: string;
  status: LeadStatus;
  createdAt: Date;
  updatedAt: Date;
};

File: backend/src/Application/Features/Leads/GetById/LeadDetailsDto.cs
using RealEstate.Domain.Enums.Leads;

namespace RealEstate.Application.Features.Leads.GetById;

public sealed record LeadDetailsDto(
    int Id,
    int AgencyId,
    int BrokerId,
    int PropertyId,
    string FullName,
    string? Email,
    string? PhoneNumber,
    string? Message,
    LeadStatus Status,
    DateTime CreatedAt,
    DateTime UpdatedAt
);